react-hot-button-component/src/hot-button.jsx

```diff
@@ -0,0 +1,44 @@
+import React from "react";
+
+class HotButton extends React.Component {
+  constructor(props){ //initiate state
+    super(props)
+    this.state ={clicks:0};
+    this.btnClicks = this.btnClicks.bind(this);
+  }
+
+  btnClicks(){//function that goes into event listener
+    this.setState({clicks: this.state.clicks+1})
+  }
+
+  render() {
+    if (this.state.clicks < 3){
+
+       return <button className="button" onClick={this.btnClicks}>Hot Button</button>
+    }
+    else if (this.state.clicks < 6){
+
+    return <button className="darkPurple button" onClick={this.btnClicks}>Hot Button</button>
+    }
+    else if (this.state.clicks < 9){
+
+       return <button className="purple button" onClick={this.btnClicks}>Hot Button</button>
+    }
+    else if (this.state.clicks < 12){
+
+       return <button className="red button" onClick={this.btnClicks}>Hot Button</button>
+    }
+    else if (this.state.clicks < 15){
+       return <button className="orange button" onClick={this.btnClicks}>Hot Button</button>
+    }
+    else if (this.state.clicks < 18){
+
+       return <button className="yellow button" onClick={this.btnClicks}>Hot Button</button>
+    }
+    else{
+
+    return <button className="white button" onClick={this.btnClicks}>Hot Button</button>
+    }
```

Comment:
Creating several of the same JSX elements is really hard to maintain, and debug. It is best to use only one return of JSX and use variables to set the parts that are going to change. For example:

```JSX
  let btnClass;
  if (this.state.clicks < 6){
    btnClass = "darkPurple";
  }
  else if (this.state.clicks < 9){
     btnClass = "purple";
  }
  else if (this.state.clicks < 12){
    btnClass = "red";
  }
  else if (this.state.clicks < 15){
    btnClass = "orange";
  }
  else if (this.state.clicks < 18){
    btnClass = "yellow";
  } else {
    btnClass ="white";
  }

  return <button className={`button ${btnClass}`} onClick={this.btnClicks}>Hot Button</button>
```